### ISSUE TITLE
Overridden operators issue

### ISSUE THREAD

**Issue by marcomaida** [BrUnOXaVIeRLeiTE/SCUE4-Plugin#1]
Hey there! very nice plugin

I found out that all overridden operators (-=, +=. ...) are broken: they do not modify the structure, they just return the value.
You can just try with:

FSafeInt test;
test = 5;
test -= 1;

Test will remain 5

**Comment by BrUnOXaVIeRLeiTE** [comment]
They were added to be used in Blueprints.
So all they do is returning a copy.
If you want to use those operators in C++ (which isn't really needed) then yes it's a problem.

**Comment by BrUnOXaVIeRLeiTE** [comment]
Fixed:  [7f090798069596af53730f28dea341b59ffdfdd2](https://github.com/BrUnOXaVIeRLeiTE/SCUE4-Plugin/commit/7f090798069596af53730f28dea341b59ffdfdd2)

11,700 lines changed.

**Comment by marcomaida** [comment]
Thank you Bruno.

I also have another thing to tell you: I had replication issues with safe
int/float (but I think all of them have this problem)

From what I understood, the problem is that reading the variable is not a
pure function anymore
Since the swap is executed also when reading, if the client reads for an
uneven number of times from one replication to another, the flag variable
becomes different from the server.

If, at the same time, the server accesses an even number of time the same
variable, the flag stays the same, and is ignored by the replication system.

Result: occasionally, the client loses variables for some time (it’s all a
game of chances). The solution for me was to eliminate the flag system.
Now, when the struct wants to access the encoded string, simply checks
which variable is null, and uses the other. This is leaner and solves all
replication issues. The only case in which this system may not work is when
both cells are null. But if you think about it, then it does not matter
where you read/write.

This solution makes the struct lighter and completely eliminates the
replication issues. I would gladly make a pull request, but my fix was done
in a very quick and dirty manner and only on safeint / safefloat, plus I
don't know if you like this radical change.

Another thing: I also found some switches missing breaks in some setter
functions. Sometimes you executed both of the flows. I did not study that
in depth, but maybe you want to check them out! :)
If you want to see the changed described above, I can send you the version
without flags of safeint and safefloat.

Nice work anyway! Have a nice day!


Bruno Xavier <notifications@github.com> schrieb am Di., 23. Juli 2019,
04:43:
